libwild/src/archive.rs

```diff
@@ -184,15 +223,51 @@ impl<'data> ArchiveContent<'data> {
         }
     }
 
-    pub(crate) fn data_range(&self) -> Range<usize> {
-        self.data_offset..self.data_offset + self.entry_data.len()
+    pub(crate) fn data_range(&self) -> Option<Range<usize>> {
+        self.entry_data
+            .map(|entry_data| self.data_offset..self.data_offset + entry_data.len())
+    }
+
+    // Parse the identifier as a reference to extended filenames
+    pub(crate) fn parse_as_thin_reference(
```

Comment:
I think to solve the TODO below, you can probably just get rid of this method. The caller of this method can just do something like:

```rust
let fname = archive_entry.identifier(extended_filenames).as_slice();
let bytes = mmap_file(Path::new(OsStr::from_bytes(fname)), false)?;
```

Reply:
Ah, I see. I never realized there was a cheaper way to construct the path other than via `str::from`!

Reply:
Done